### PR TITLE
Request counter

### PR DESCRIPTION
**Pull request by volovyks**

### CODE REVIEW

File: contract/src/lib.rs [hunk 1]
```diff
@@ -49,7 +49,8 @@ pub enum ProtocolContractState {
 #[derive(BorshDeserialize, BorshSerialize, PanicOnDefault)]
 pub struct MpcContract {
     protocol_state: ProtocolContractState,
-    pending_requests: TreeMap<[u8; 32], Option<(String, String)>>,
+    pending_requests: LookupMap<[u8; 32], Option<(String, String)>>,
+    request_counter: u32,
```
Comment (ppca) [inline contract/src/lib.rs:53]: this will break contract I'm guessing? since when it was initialzed it did not have this field.

Comment (volovyks) [reply contract/src/lib.rs:53]: It depends on how borsh deserialization works, I have not tested it. I belive it should be initialized on the first `sign` call.

Comment (ChaoticTempest) [reply contract/src/lib.rs:53]: here's an example of how we can migrate state: https://github.com/near/near-sdk-rs/blob/1408127feefb37e7f84eba25984af18af46cd91d/examples/test-contract/src/lib.rs#L20

Comment (ChaoticTempest) [reply contract/src/lib.rs:53]: so we need a new function similar to `clean` but ignores the state

Comment (ppca) [reply contract/src/lib.rs:53]: Let me try redeploy the dev contract.

Comment (ppca) [reply contract/src/lib.rs:53]: yep. I got this error ```
kind: {
    ExecutionError: 'Smart contract panicked: panicked at /Users/xiangyiz/.cargo/registry/src/index.crates.io-6f17d22bba15001f/near-sdk-5.0.0-alpha.2/src/environment/env.rs:921:46:\n' +
      'Cannot deserialize the contract state.: Custom { kind: InvalidData, error: "Unexpected length of input" }'
  },
```

Comment (ppca) [reply contract/src/lib.rs:53]: all 3 contracts migrated

Comment (volovyks) [reply contract/src/lib.rs:53]: k, migration function deleted

**Review by ppca — APPROVED** [review]

**Review by ChaoticTempest — APPROVED** [review]
We should move to versioned contracts so this process is a lot easier: https://github.com/near/mpc-recovery/issues/545

**Pull request merged**